consensus/src/consensus/mod.rs

```diff
@@ -1372,25 +1375,18 @@ impl ConsensusApi for Consensus {
         pruning_meta_write.utxo_set.clear().unwrap();
     }
 
-    fn verify_is_pruning_sample(&self, pruning_candidate: Hash) -> ConsensusResult<()> {
-        if pruning_candidate == self.config.genesis.hash {
-            return Ok(());
-        }
+    fn verify_is_pruning_sample(&self, pruning_candidate: Hash) -> bool {
         let Ok(candidate_ghostdag_data) = self.get_ghostdag_data(pruning_candidate) else {
-            return Err(ConsensusError::General("pruning candidate missing ghostdag data"));
+            return false;
```

Comment:
This should be an error, not false, since we can't determine in this case.

I suggest to do one of the following:

1. Revert this function to the way it was before.
2. Remove it and inline its use in its only caller site, `get_and_verify_path_to_new_pruning_point`